### PR TITLE
Updated CMakeLists.txt slightly to allow building this project in Android Studio with the latest Android NDK

### PR DESCRIPTION
**Pull request by l3utterfly**
Since llama.cpp can get reasonable inference speeds on latest flagship Android devices, I have been experimenting linking it into an Android app and creating a JNI around it etc.

I found some compiler flags needed to be modified to make it build correctly. I believe this is because the latest Android NDK uses Clang instead of GCC.

This is what I had to do to make it build successfully.

After these changes, the project can be directly linked using Android Studio's GUI and built normally.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -323,7 +323,12 @@ if (${CMAKE_SYSTEM_PROCESSOR} MATCHES "arm" OR ${CMAKE_SYSTEM_PROCESSOR} MATCHES
         if (${CMAKE_SYSTEM_PROCESSOR} MATCHES "aarch64")
             # Apple M1, M2, etc.
             # Raspberry Pi 3, 4, Zero 2 (64-bit)
-            add_compile_options(-mcpu=native)
+
+            # latest Android NDK uses CLang now, which does not support the native flag
+            if(DEFINED ANDROID_NDK)
```
Comment (ggerganov) [inline CMakeLists.txt:328]:
```suggestion
            if (DEFINED ANDROID_NDK)
```

File: CMakeLists.txt [hunk 2]
```diff
@@ -336,6 +341,10 @@ if (${CMAKE_SYSTEM_PROCESSOR} MATCHES "arm" OR ${CMAKE_SYSTEM_PROCESSOR} MATCHES
         if (${CMAKE_SYSTEM_PROCESSOR} MATCHES "armv8")
             # Raspberry Pi 3, 4, Zero 2 (32-bit)
             add_compile_options(-mfp16-format=ieee -mno-unaligned-access)
+
+            if(DEFINED ANDROID_NDK)
+            add_compile_options(-march=armv8.4a+dotprod)
+            endif()
```
Comment (ggerganov) [inline CMakeLists.txt:347]:
```suggestion

            if(DEFINED ANDROID_NDK)
                add_compile_options(-march=armv8.4a+dotprod)
            endif()
```

**Review by ggerganov — CHANGES_REQUESTED** [review]

File: CMakeLists.txt [hunk 3]
```diff
@@ -233,18 +233,14 @@ if (LLAMA_K_QUANTS)
 endif()
 
 if (LLAMA_CLBLAST)
-    find_package(CLBlast)
-    if (CLBlast_FOUND)
-        message(STATUS "CLBlast found")
+    # build CLBlast from source
+    add_subdirectory(../CLBlast ${CMAKE_CURRENT_BINARY_DIR}/clblast)
```
Comment (ggerganov) [inline CMakeLists.txt:237]: This change is not OK.
You probably have to add some custom CMake option to be able to optionally select CLBlast from source

**Review by ggerganov — CHANGES_REQUESTED** [review]

File: CMakeLists.txt (hunk 3 above)
Comment (l3utterfly) [reply CMakeLists.txt:237]: Alright, will do

**Pull request closed without merging**